src/r_plane.c

```diff
@@ -376,152 +383,258 @@ static void R_MakeSpans(int x, unsigned int t1, unsigned int b1, unsigned int t2
+            dc_source = R_GetFireColumn((viewangle + xtoskyangle[x])
+                                        >> ANGLETOSKYSHIFT);
+            colfunc();
+        }
+    }
+}
+
+static void DrawSkyTex(visplane_t *pl, skytex_t *skytex)
 {
-  register int x;
-  if (pl->minx <= pl->maxx)
-  {
-    if (pl->picnum == skyflatnum || pl->picnum & PL_SKYFLAT)  // sky flat
-      {
-	int texture;
-	angle_t an, flip;
-	boolean vertically_scrolling = false;
-	boolean stretch;
-
-	// killough 10/98: allow skies to come from sidedefs.
-	// Allows scrolling and/or animated skies, as well as
-	// arbitrary multiple skies per level without having
-	// to use info lumps.
-
-	an = viewangle;
-
-	if (pl->picnum & PL_SKYFLAT)
-	  { 
-	    // Sky Linedef
-	    const line_t *l = &lines[pl->picnum & ~PL_SKYFLAT];
-
-	    // Sky transferred from first sidedef
-	    const side_t *s = *l->sidenum + sides;
-
-	    if (s->baserowoffset - s->oldrowoffset)
-	      vertically_scrolling = true;
-
-	    // Texture comes from upper texture of reference sidedef
-	    texture = texturetranslation[s->toptexture];
-
-	    // Horizontal offset is turned into an angle offset,
-	    // to allow sky rotation as well as careful positioning.
-	    // However, the offset is scaled very small, so that it
-	    // allows a long-period of sky rotation.
-
-	    an += s->textureoffset;
-
-	    // Vertical offset allows careful sky positioning.
-
-	    dc_texturemid = s->rowoffset - 28*FRACUNIT;
-
-	    // We sometimes flip the picture horizontally.
-	    //
-	    // Doom always flipped the picture, so we make it optional,
-	    // to make it easier to use the new feature, while to still
-	    // allow old sky textures to be used.
-
-	    flip = l->special==272 ? 0u : ~0u;
-	  }
-	else 	 // Normal Doom sky, only one allowed per level
-	  {
-	    dc_texturemid = skytexturemid;    // Default y-offset
-	    texture = skytexture;             // Default texture
-	    flip = 0;                         // Doom flips it
-	  }
-
-        // Sky is always drawn full bright, i.e. colormaps[0] is used.
-        // Because of this hack, sky is not affected by INVUL inverse mapping.
-	//
-	// killough 7/19/98: fix hack to be more realistic:
-
-	if (STRICTMODE_COMP(comp_skymap) || !(dc_colormap[0] = dc_colormap[1] = fixedcolormap))
-	  dc_colormap[0] = dc_colormap[1] = fullcolormap;          // killough 3/20/98
-
-        dc_texheight = textureheight[texture]>>FRACBITS; // killough
-        dc_iscale = skyiscale;
-
-        // [FG] stretch short skies
-        stretch = (stretchsky && dc_texheight < 200);
-        if (stretch || !vertically_scrolling)
+    int texture = R_TextureNumForName(skytex->name);
+
+    dc_colormap[0] = dc_colormap[1] = fullcolormap;
+
+    dc_texturemid = skytex->mid * FRACUNIT;
+    dc_texheight = textureheight[texture] >> FRACBITS;
+    dc_iscale = FixedMul(skyiscale, skytex->scaley);
+
+    dc_texturemid += skytex->curry;
+
+    angle_t an = viewangle + FixedToAngle(skytex->currx);
```

Comment:
Scrolling/scaling works, but not exactly as in the KEX port. R&R implementation is complicated: https://github.com/GooberMan/rum-and-raisin-doom/blob/4af881489bda0b26f12744de05dba7a571f2faa9/src/doom/r_sky.cpp#L429

Test WAD: [sky.zip](https://github.com/user-attachments/files/16982403/sky.zip)